vendor/bat-native-ledger/src/bat/ledger/internal/wallet_provider/bitflyer/connect_bitflyer_wallet.cc

```diff
@@ -0,0 +1,108 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/wallet_provider/bitflyer/connect_bitflyer_wallet.h"
+
+#include <utility>
+
+#include "base/containers/contains.h"
+#include "base/strings/string_number_conversions.h"
+#include "bat/ledger/global_constants.h"
+#include "bat/ledger/internal/bitflyer/bitflyer.h"
+#include "bat/ledger/internal/bitflyer/bitflyer_util.h"
+#include "bat/ledger/internal/common/random_util.h"
+#include "bat/ledger/internal/endpoint/bitflyer/bitflyer_server.h"
+#include "bat/ledger/internal/endpoints/post_connect/bitflyer/post_connect_bitflyer.h"
+#include "bat/ledger/internal/endpoints/request_for.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "bat/ledger/internal/logging/event_log_keys.h"
+#include "crypto/sha2.h"
+
+using ledger::endpoints::PostConnectBitflyer;
+using ledger::endpoints::RequestFor;
+using ledger::wallet_provider::ConnectExternalWallet;
+
+namespace ledger::bitflyer {
+
+ConnectBitFlyerWallet::ConnectBitFlyerWallet(LedgerImpl* ledger)
+    : ConnectExternalWallet(ledger),
+      bitflyer_server_(std::make_unique<endpoint::BitflyerServer>(ledger)) {}
+
+ConnectBitFlyerWallet::~ConnectBitFlyerWallet() = default;
+
+const char* ConnectBitFlyerWallet::WalletType() const {
+  return constant::kWalletBitflyer;
+}
+
+void ConnectBitFlyerWallet::Authorize(
+    OAuthInfo&& oAuthInfo,
```

Comment:
Nit: `oauth_info` (it's done in a few different places throughout the PR)

Reply:
Saw this one too

Reply:
Lol, good catch, thanks!